### ISSUE TITLE
pytorch

### ISSUE THREAD

**Issue by ifangcheng** [thu-ml/zhusuan#63]
Can zhusuan be applied within pytorch framework？
 OR will it support pytorch in future？:-D

**Comment by thjashin** [comment]
Maybe not. I'm personally preferring Tensorflow for several reasons. I may list them here in case other people have interests.

1. Static graphs will be much easier to optimize. Now TF may be reported slow under some settings. But it will soon change. The [XLA compiler](https://www.tensorflow.org/performance/xla/) is not merged yet. Once done one is able to select optimization options when executing graphs (`tf.OptimizerOptions.ON_1`). 

2. Dynamic graph style primitives are not that hard to support. Actually Tensorflow has made attempts towards this. See the [tf.contrib.eager](https://www.quora.com/Can-one-do-imperative-programming-in-TensorFlow) module. It will be merged into core of TF in future. I'm expecting more such as defining a function with dynamic style of code, and adding a decorator to transform to a static graph. And TF eager is actually showing some examples like this.

3. Distributed execution is better support in TF. Pytorch's distributed module is still only some abstractions over common protocols.

4. Many other reasons such as larger community, not good enough but complete documents, and

**Comment by ifangcheng** [comment]
Many thanks for reply.
Just wondering is there a model zoo for the–state–of–the–art DGMs developed in your group? 
If there is, it would be much easier to follow the published papers.

**Comment by renjinhao** [comment]
When I run the bayesian_nn.py, the results show that  "from examples import conf, ImportError: No module named 'examples'". However, I have installed the examples by "pip install ".[examples]"",. I wonder whether there is anything others need to be done before running it??

**Comment by thjashin** [comment]
@renjinhao Since this is unrelated to this issue, can you raise your question in a new issue?

**Comment by thjashin** [comment]
@ifangcheng In general any paper published with code should be released under the thu-ml organization. If you don't find them, that's probably due to it is still not ready for a release.

**Comment by ifangcheng** [comment]
@thjashin   I noted a DGM project (mmdcgm ) under the thu-ml organization, However, it seems that it is NOT  developed based on zhusuan ? 
Any zhusuan version about it? or will release in future?

**Comment by thjashin** [comment]
@ifangcheng This is an early work before we have zhusuan, so unfortunately not :(
